docs/how-to/integrate-with-s3.md

```diff
@@ -0,0 +1,181 @@
+```
+
+And run the minio server (username: `minio`, password: `minio123`)
+
+```
+docker run --network host -p 9000:9000 -p 9001:9001 -e "MINIO_ROOT_USER=minio" -e "MINIO_ROOT_PASSWORD=minio123" quay.io/minio/minio server /data --console-address ":9001"
+```
+
+You should be able to login to minio at `http://localhost:9001`.
+
+Login using username `minio` and password `minio123`. Click "Create Bucket"
+and call it `mquery`.
+
+### 3. Enable S3 plugin
+
+Open the Mquery config file (`/opt/mquery/src/config.py` in the example installation):
+
+```
+vim /opt/mquery/src/config.py
+```
+
+Change `PLUGINS` key to:
+
+```
+PLUGINS = ["plugins.s3_plugin:S3Plugin"]
+```
+
+And exit vim with `[esc]:x[enter]`.
+
+Restart mquery workers and the web interface.
+
+### 4. Configure the plugin
+
+If you did it correctly, workers should print the following message:
+
+```
+[12/01/2023 00:23:40][ERROR] Failed to load S3Plugin plugin
+Traceback (most recent call last):
+  File "/opt/mquery/src/plugins/__init__.py", line 49, in __init__
+    active_plugins.append(plugin_class(db, plugin_config))
+  File "/opt/mquery/src/plugins/s3_plugin.py", line 24, in __init__
+    super().__init__(db, config)
+  File "/opt/mquery/src/metadata.py", line 28, in __init__
+    raise KeyError(
+KeyError: "Required configuration key 's3_url' is not set"
+```
+
+Navigate to the Mquery config page at http://localhost/config. You should see
+the plugin configuration there. Set all the fields:
+
+* `s3_url` to your S3 url - in our example this will be `your_ip:9000` (for
+   example `1.2.3.4:9000`, or just `localhost:9000`).
+   Remember, do not add `http://`.
+* `s3_bucket` to your s3 bucket name - in our example `mquery`.
+* `s3_access_key` and `s3_secret_key` to your s3 credentials for mquery.
+  Create key pair in minio if you don't have one already.
+* `s3_secure` to `false` (in our example - you probably want HTTPs in production).
+
+At this point, workers should be able to load plugins correctly.
+
+### 5. Index your files
+
+Good news - that's everything you need for querying! Bad news - you still
+need to index your files.
+
+To do this, you will need to run a dedicated s3 indexing script. If your ursadb
+instance is on another server than your workers, you must run this script on
+the ursadb server (to be more precise, you need a shared storage with Ursadb.
+The easiest way to obtain this is to run on the same disk).
+
+Go to the mquery directory and execute the following (fix the parameters
+depending on your use case):
+
+```
+cd /opt/mquery/src/
+source /opt/mquery/venv/bin/activate
+python3 -m utils.s3index \
+    --workdir /root/mquery_tmp \
+    --s3-url localhost:9000 \
+    --s3-secret-key YOUR-SECRET-KEY \
+    --s3-access-key YOUR-ACCESS-KEY \
+    --s3-bucket mquery \
+    --s3-secure 0
+```
+
+`--workdir` is used to specify directory where the samples are temporarily downloaded. This is important, because this will be the path that ursadb sees
```

Comment:
```suggestion
`--workdir` is used to specify directory where the samples are temporarily downloaded. This is important, because this will be the path that UrsaDB sees
```